### PR TITLE
fix HTTPS m3u8 file not loading

### PR DESCRIPTION
**Pull request by AhmedAbouelkher**
Added a configuration setting to the player to set [demuxer-lavf-o](https://mpv.io/manual/stable/#options-demuxer-lavf-o) with the value `protocol_whitelist=[file,tcp,tls,http,https]` this property allows the library to parse M3U8 files with the specified protocols, which should resolve issues with playing HTTPS M3U8 files.

### Dart example with using  `protocolWhitelist`
```dart
final Player player = Player(
  configuration: const PlayerConfiguration(
    logLevel: MPVLogLevel.warn,
    protocolWhitelist: ["file", "tcp", "tls", "http", "https", "crypto","data"],
  ),
)
```
### Test File from [omagestream](https://www.omegastream.net/Home) service

```m3u8
#EXTM3U
#EXT-X-VERSION:3
#EXT-X-INDEPENDENT-SEGMENTS

#EXT-X-STREAM-INF:BANDWIDTH=674554,CODECS="avc1.640015,mp4a.40.2",RESOLUTION=426x240,FRAME-RATE=30,AUDIO="audio2"
https://api.omegastream.net/stream/company/processed-enterprise-media/bd413f82/video/57721636-0247-4588-8772-b2ec036504e0/playlist_1.m3u8?s=58c953dcbcfda4dd29e9cafa2c0a38a21964fef7

#EXT-X-STREAM-INF:BANDWIDTH=994337,CODECS="avc1.64001E,mp4a.40.2",RESOLUTION=640x360,FRAME-RATE=30,AUDIO="audio2"
https://api.omegastream.net/stream/company/processed-enterprise-media/bd413f82/video/57721636-0247-4588-8772-b2ec036504e0/playlist_2.m3u8?s=58c953dcbcfda4dd29e9cafa2c0a38a21964fef7

#EXT-X-STREAM-INF:BANDWIDTH=1518890,CODECS="avc1.64001F,mp4a.40.2",RESOLUTION=854x480,FRAME-RATE=30,AUDIO="audio2"
https://api.omegastream.net/stream/company/processed-enterprise-media/bd413f82/video/57721636-0247-4588-8772-b2ec036504e0/playlist_3.m3u8?s=58c953dcbcfda4dd29e9cafa2c0a38a21964fef7

#EXT-X-STREAM-INF:BANDWIDTH=2666804,CODECS="avc1.64001F,mp4a.40.2",RESOLUTION=1280x720,FRAME-RATE=30,AUDIO="audio2"
https://api.omegastream.net/stream/company/processed-enterprise-media/bd413f82/video/57721636-0247-4588-8772-b2ec036504e0/playlist_4.m3u8?s=58c953dcbcfda4dd29e9cafa2c0a38a21964fef7

#EXT-X-MEDIA:TYPE=AUDIO,GROUP-ID="audio2",NAME="5",AUTOSELECT=YES,URI="https://api.omegastream.net/stream/company/processed-enterprise-media/bd413f82/video/57721636-0247-4588-8772-b2ec036504e0/playlist_5.m3u8?s=58c953dcbcfda4dd29e9cafa2c0a38a21964fef7",CHANNELS="2"
```

### Example using MPV command line tool
```sh
mpv --demuxer-lavf-o="protocol_whitelist=[file,tcp,tls,http,https]" example.m3u8
```

Credits: @zezo357 helped alot ♥️

### CODE REVIEW

**Comment by solsticedhiver** [comment]
Not a dev here, but do you have an example of a failing https m3u8?

Is it only with omegastream?

Because I did not had the problem. with the wrong protocol. It fails sometimes for over reason.

**Comment by zezo357** [comment]
> Not a dev here, but do you have an example of a failing https m3u8?
> 
> Is it only with omegastream?
> 
> Because I did not had the problem. with the wrong protocol. It fails sometimes for over reason.

It's fails when playing a m3u8 local file like the mentioned in PR , and mostly it's https not being allowed


It works if it's a m3u8 url

**Comment by solsticedhiver** [comment]
oh yes. sorry

**Comment by zezo357** [comment]
> oh yes. sorry
> 
> Is the following not working?
> 
> ```dart
> final player = Player();
> if (player.platform is libmpvPlayer) {
>   await (player.platform as dynamic).setProperty('demuxer-lavf-o', 'protocol_whitelist=[file,tcp,tls,http,https]');
> }
> ```
> 
> adapted from https://github.com/alexmercerind/media_kit/issues/129#issuecomment-1508019709

Yes this was added in the player configuration. In this pr.

**Comment by solsticedhiver** [comment]
yes. I saw.
The question is, is this needed to hard-code it, in the player config? or not?

Given, you can add it yourself, before playing the video, if this is what kind of source you are expecting to play.

This was what I meant.

**Comment by zezo357** [comment]
> yes. I saw.
> The question is, is this needed to hard-code it, in the player config? or not?
> 
> Given, you can add it yourself, before playing the video, if this is what kind of source you are expecting to play.
> 
> This was what I meant.

For me yes, I added it to player before playing, but me and ahmed thought some settings should be added to player configurations to make it easier for developers

**Review by alexmercerind — APPROVED** [review]
LGTM (+nits).

May you please replace `"` with `'` as we are using them throughout our source code.
Also, we enforce `dart format` / format on save as supported by your code editor.

**Comment by alexmercerind** [comment]
> Same for HTTP headers, thought it may be better to set it in `Media`.

Yeah, it is in the plans.

File: media_kit/lib/src/platform_player.dart [hunk 1]
```diff
@@ -98,6 +98,13 @@ class PlayerConfiguration {
   /// Default: `32` MB or `32 * 1024 * 1024` bytes.
   final int bufferSize;
 
+  /// Sets the list of allowed protocols for libmpv backend.
+  ///
+  /// Default: `["file", "tcp", "tls", "http", "https", "crypto","data"]`.
```
Comment (alexmercerind) [inline media_kit/lib/src/platform_player.dart:103]:
```suggestion
  /// Default: `["file", "tcp", "tls", "http", "https", "crypto", "data"]`.
```

**Comment by zezo357** [comment]
> > Same for HTTP headers, thought it may be better to set it in `Media`.
> 
> Yeah, it is in the plans.

Will try to make a pr for it.

**Pull request merged**